astropy/units/format/ogip.py

```diff
@@ -98,11 +98,11 @@ def _units(cls) -> dict[str, UnitBase]:
     def _lexer(cls) -> Lexer:
         tokens = cls._tokens
 
-        t_DIVISION = r"/"
+        t_DIVISION = "[ \t]*/[ \t]*"
         t_OPEN_PAREN = r"\("
         t_CLOSE_PAREN = r"\)"
         t_WHITESPACE = "[ \t]+"
-        t_STARSTAR = r"\*\*"
+        t_POWER = r"\*\*"
         t_STAR = r"\*"
```

Comment:
I wondered if it would sense to define a `t_MULTIPLICATION` (or `t_PRODUCT`) which includes arbitrary whitespace similar to your `t_DIVISION`. It would simplify `p_product` (removing three lines).

Trying it on this PR, I see this would require `t_MULTIPLICATION = "[ \t]*\\*(?!\\*)[ \t]*"`, i.e., one has to explicitly avoid also matching on `**`. This is a bit annoying, and if one were instead to enforce order by defining the power as a function, the advantage is really gone. So, perhaps rather stick with the slight asymmetry you have here!

Reply:
The changes to `t_DIVISION` allowed removing the `p_division` parser rule entirely. The regex you are suggesting for `t_MULTIPLICATION` is still not good enough to replace the parser rule and it is not obvious if simplifying the parser rule is a good thing to do if the tradeoff is making the token much more complicated.

I'm not worried about the code for unit division and multiplication being different. In OGIP `/` is only used for division and division requires `/`. In contrast `*` is used for both multiplication and exponentiation and multiplication does not require `*`. These differences are not an artifact our implementation, they are inherent to the standard.

Reply:
OK, let's merge it as is -- a definite improvement!